pkg/mail/gmailauth.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package mail
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/smtp"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+)
+
+// GmailAuth is XOAUTH2 implementation of smtp.Auth interface
+type GmailAuth struct {
+	Mail string
```

Comment:
Is it `From`, `To`, authentication address etc.? This doesn't give enough information.